var/spack/repos/builtin/packages/goma/package.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Goma(CMakePackage):
+    """A Full-Newton Finite Element Program for Free and Moving Boundary Problems with
+       Coupled Fluid/Solid Momentum, Energy, Mass, and Chemical Species Transport """
+
+    homepage = "https://www.gomafem.com"
+    url = "https://github.com/goma/goma/archive/v7.0.0.tar.gz"
+    git = "https://github.com/goma/goma.git"
+
+    maintainers = ['wortiz']
+
+    version('7.0.0', commit='5166896f273e5853e1f32885e20f68317b24979c')
+    version('main', branch='main')
+    version('develop', branch='develop')
+
+    variant('MDE', default=27, description="Set internal maximum DOF per element")
+    variant('MAX_PROB_VAR', default=15, description="Set internal maximum number of active equations")
+    variant('MAX_CONC', default=4, description="Set internal maximum number of species")
+    variant('MAX_EXTERNAL_FIELD', default=4, description="Set internal maximum number of external fields")
```

Comment:
Two comments:
1. We usually have lowercase variants, separated by an underscore (since `-` can be a synonym to `~` in some contexts)
2. Can we define a set of allowed values for these 4 variants?

Reply:
I think I've addressed this. There isn't any way to specify a range of numbers allowed is there? I did need to change them to strings to allow the value checker to work but cmake results seem fine.

Reply:
Thanks!